busybox: update to 1.35.0

hnyman opened the pull request:
Update busybox to 1.35.0

* refresh patches

Config refresh:

Refresh commands, run after busybox is first built once:

```
cd package/utils/busybox/config/
../convert_menuconfig.pl ../../../../build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/busybox-default/busybox-1.35.0
cd ..
./convert_defaults.pl ../../../build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/busybox-default/busybox-1.35.0/.config > Config-defaults.in
```
Manual edits needed after config refresh:

* Config-defaults.in: OpenWrt config symbol IPV6 logic applied to
  BUSYBOX_DEFAULT_FEATURE_IPV6

* Config-defaults.in: OpenWrt configTARGET_bcm53xx logic applied to
  BUSYBOX_DEFAULT_TRUNCATE (commit 547f1ec)

* Config-defaults.in: OpenWrt logic applied to
  BUSYBOX_DEFAULT_LOGIN_SESSION_AS_CHILD (commit dc92917)

* config/editors/Config.in: Add USE_GLIBC dependency to
  BUSYBOX_CONFIG_FEATURE_VI_REGEX_SEARCH (commit f141090)

* config/shell/Config.in : change at "Options common to all shells" the symbol
  SHELL_ASH --> BUSYBOX_CONFIG_SHELL_ASH
  (discussion in http://lists.openwrt.org/pipermail/openwrt-devel/2021-January/033140.html
  Apparently our script does not see the hidden option while
  prepending config options with "BUSYBOX_CONFIG_" which leads to a
  missed dependency when the options are later evaluated.)

* Edit Config.in files by adding quotes to sourced items in
  config/Config.in, config/networking/Config.in and config/util-linux/Config.in (commit 1da014f)

-----

Initial version, briefly run-tested with R7800 so far. 
(I also tested further sysupgrade...)

Note that there are some config definition segment movements in upstream, which makes the patchset to look larger than the actual end-changes are.

I haven't yet tested this extensively, but thought to publish it already now for testing.

Busybox 1.35.0 is still an "unstable" release, so waiting a few days for possible quick upstream fixes might be sensible.

rsalvaterra commented:
Thanks, @hnyman! On a cursory look, this seems fine. I'll give it a spin on my Redmi AC2100.

rsalvaterra reviewed (approved):
Ok, had a bit more time to look at the changes and they look good to me. Also tested on my Redmi AC2100, without issues.

hnyman commented:
I have also tested this on ipq806x/R7800 and mt7622/E8450 and have not noticed problems.

So far there haven't been any upstream fixes, so likely this is ok.

hauke commented on package/utils/busybox/Config-defaults.in:
```diff
@@ -2544,6 +2568,9 @@ config BUSYBOX_DEFAULT_TELNETD
 	default n
+config BUSYBOX_DEFAULT_FEATURE_TELNETD_PORT_DEFAULT
+	int
+	default 0
```
I would prefer if this is set to port 23 instead of 0. 
0 is also not in range of `BUSYBOX_CONFIG_FEATURE_TELNETD_PORT_DEFAULT`

hnyman replied:
Changed.

hauke commented on package/utils/busybox/Config-defaults.in:
```diff
@@ -2265,6 +2286,9 @@ config BUSYBOX_DEFAULT_DNSDOMAINNAME
 	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_PORT_DEFAULT
+	int
+	default 0
```
I would prefer 80 for this value.

hnyman replied:
Changed.

hnyman committed:
busybox: update to 1.35.0

Update busybox to 1.35.0

* refresh patches

Config refresh:

Refresh commands, run after busybox is first built once:

cd package/utils/busybox/config/
../convert_menuconfig.pl ../../../../build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/busybox-default/busybox-1.35.0
cd ..
./convert_defaults.pl ../../../build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/busybox-default/busybox-1.35.0/.config > Config-defaults.in

Manual edits needed after config refresh:

* Config-defaults.in: OpenWrt config symbol IPV6 logic applied to
  BUSYBOX_DEFAULT_FEATURE_IPV6

* Config-defaults.in: OpenWrt configTARGET_bcm53xx logic applied to
  BUSYBOX_DEFAULT_TRUNCATE (commit 547f1ec)

* Config-defaults.in: OpenWrt logic applied to
  BUSYBOX_DEFAULT_LOGIN_SESSION_AS_CHILD (commit dc92917)

* config/editors/Config.in: Add USE_GLIBC dependency to
  BUSYBOX_CONFIG_FEATURE_VI_REGEX_SEARCH (commit f141090)

* config/shell/Config.in : change at "Options common to all shells" the symbol
  SHELL_ASH --> BUSYBOX_CONFIG_SHELL_ASH
  (discussion in http://lists.openwrt.org/pipermail/openwrt-devel/2021-January/033140.html
  Apparently our script does not see the hidden option while
  prepending config options with "BUSYBOX_CONFIG_" which leads to a
  missed dependency when the options are later evaluated.)

* Edit Config.in files by adding quotes to sourced items in
  config/Config.in, config/networking/Config.in and config/util-linux/Config.in (commit 1da014f)

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>

hnyman commented:
I changed the numerical defaults of those two disabled features that @hauke flagged.

hauke commented:
Thank you for the patch, I applied it to master in a54b406ccb7b29a86e43e5535b86efede52157e6.
